tests/test_process.py

```diff
@@ -0,0 +1,26 @@
+import asyncio
+from unittest.mock import AsyncMock, patch
+
+from meltano.edk.process import Invoker
+
+
+def test_exec():
+    """Verify that the Invoker._exec method works as expected."""
+
+    async def _test_exec():
+        inv = Invoker("echo", cwd="/tmp", env={"FOO": "BAR"})
+        with patch("asyncio.create_subprocess_exec") as mock:
+            mock.return_value = AsyncMock()
+            await inv._exec("sub_command", ["arg1", "arg2"])
+            mock.assert_called_once_with(
+                "echo",
+                "sub_command",
+                "arg1",
+                "arg2",
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+                cwd="/tmp",
+                env={"FOO": "BAR"},
+            )
+
+    asyncio.run(_test_exec())
```

Comment:
Starting to add some tests since i clearly keep messing this up :)
 
Guessing we'll probably wanna pull pytest-asyncio at some point.

Reply:
@pandemicsyn what do you think would be the threshold to make `pytest-asyncio` a necessity?

Reply:
@edgarrmondragon probably once we have additional async method's beyond just _exec that we have to test.  Right now `run_and_log` is actually also just doing a:

```
result = asyncio.run(self._exec(sub_command, *args))
```

Just felt like overkill adding pytest-asyncio just so I could use @pytest.mark.async for testing the lone method.  

If its preferable, I could also actually just rewrite the test , to test this path via a call to `run_and_log`. Then the test wouldn't really have to be async aware (beyond using AsyncMock).

Reply:
> If its preferable, I could also actually just rewrite the test , to test this path via a call to run_and_log. Then the test wouldn't really have to be async aware (beyond using AsyncMock).

Probably not, and once we have a handful of tests depending on asyncio makes more sense 👍